src/include/network/services/routing.rb

```diff
@@ -258,7 +271,7 @@ def RoutingEditDialog(id, entry, devs)
         end
         route = Builtins.add(route, val)
         val = Convert.to_string(UI.QueryWidget(Id(:genmask), :Value))
-        if val != "-" && val != "0.0.0.0" && !Netmask.Check(val)
+        if val != "-" && val != "0.0.0.0" && !valid_netmask?(val)
```

Comment:
bcs the input value need not be only netmask but also prefix. I agree that ```valid_netmask?``` is most probably not the best name here. I've used "netmask" as a reference to the column name which is being validated.